src/solvers/dg.jl

```diff
@@ -638,7 +638,7 @@ end
     # since LoopVectorization does not support `ForwardDiff.Dual`s. Hence, we use
     # optimized `PtrArray`s whenever possible and fall back to plain `Array`s
     # otherwise.
-    if _PREFERENCE_POLYESTER && LoopVectorization.check_args(u_ode)
+    if _PREFERENCE_THREADING === :polyester && LoopVectorization.check_args(u_ode)
```

Comment:
```suggestion
    @static if _PREFERENCE_THREADING === :polyester && LoopVectorization.check_args(u_ode)
```

Reply:
@vchuravy You recommended `@static` here, which was added and removed again.

Reply:
Ah I think the challenge is that we want `_PREFERENCE_THREADING === :polyester` to be static and the `check_args` to be not. If the compiler does it job, this should be always `true` or `false`. The `@static` is defensive coding here.

Reply:
So it should be nested like

```julia
@static if _PREFERENCE_THREADING === :polyester
    if LoopVectorization.check_args(u_ode)
         ....
    end
end
```

?

Reply:
Yeah with the annoyance that you want to hit the else branch in both cases.

Reply:
Ok, so these parts can stay as they are?